README.md

```diff
@@ -1,43 +1,169 @@
+- 24 cores
+- 12 storage devices
+- 10 GbE card
 
-```bash
-$ juju deploy zookeeper-k8s -n 3
-$ juju deploy kafka-k8s -n 3
+## Usage
+
+### Basic usage
+
+The Kafka and ZooKeeper operators can both be deployed as follows:
+```shell
+$ juju deploy zookeeper-k8s --channel latest/edge -n 5
+$ juju deploy kafka-k8s --channel latest/edge -n 3
+```
+
+After this, it is necessary to connect them:
+```shell
 $ juju relate kafka-k8s zookeeper-k8s
 ```
 
-## A fast and fault-tolerant, real-time event streaming platform!
+To watch the process, `juju status` can be used. Once all the units show as `active|idle` the credentials to access a broker can be queried with:
+```shell
+juju run-action kafka-k8s/leader get-admin-credentials --wait 
+```
+
+### Replication
+#### Scaling application
+The charm can be scaled using `juju scale-application` command.
+```shell
+juju scale-application kafka-k8s <num_of_brokers_to_scale_to>
+```
+
+This will add or remove brokers to match the required number. To scale a deployment with 3 kafka units to 5, run:
+```shell
+juju scale-application kafka-k8s 5
+```
 
-Manual, Day 2 operations for deploying and operating Apache Kafka, topic creation, client authentication, ACL management and more are all handled automatically using the [Juju Operator Lifecycle Manager](https://juju.is/docs/olm).
+### Password rotation
+#### Internal operator user
+The operator user is used internally by the Charmed Kafka K8s Operator, the `set-password` action can be used to rotate its password.
+```shell
+# to set a specific password for the operator user
+juju run-action kafka-k8s/leader set-password password=<password> --wait
 
-### Key Features
-- SASL/SCRAM auth for Broker-Broker and Client-Broker authenticaion enabled by default.
-- Access control management supported with user-provided ACL lists.
-- Fault-tolerance, replication and high-availability out-of-the-box.
-- Streamlined topic-creation through [Juju Actions](https://juju.is/docs/olm/working-with-actions) and [application relations](https://juju.is/docs/olm/relations)
+# to randomly generate a password for the operator user
+juju run-action kafka-k8s/leader set-password --wait
+```
 
+### Storage support
 
-### Checklist
+Currently, the Charmed Kafka K8s Operator makes use of a 10GB storage mount, tied to a [Kubernetes PVC](https://kubernetes.io/docs/concepts/storage/persistent-volumes/).
 
-- [x] Super-user creation
-- [x] Inter-broker auth
-- [x] Horizontally scale brokers
-- [x] Username/Password creation for related applications
-- [ ] Automatic topic creation with associated user ACLs
-- [ ] Partition rebalancing during broker scaling
-- [ ] Rack awareness support
-- [ ] Persistent storage support with [Juju Storage](https://juju.is/docs/olm/defining-and-using-persistent-storage)
-- [ ] TLS/SSL support
+This is used for logs storage, mounted on `/logs/kafka`
+
+## Relations
+
+Supported [relations](https://juju.is/docs/olm/relations):
+
+#### `data-integrator` interface:
```

Comment:
I think the interface name should be kafka-client. This interface is implemented in the data-integrator.
Suggestion: maybe you can copy/paste two sentences to add a description for the data-integrator